SocketRocket/SRWebSocket.h

```diff
@@ -45,11 +45,16 @@ extern NSString *const SRWebSocketErrorDomain;
 // It will be nil until after the handshake completes.
 @property (nonatomic, readonly, copy) NSString *protocol;
 
+// Should not be changed after opening websocket
+@property (nonatomic) BOOL allowSelfSignedCertificates;
+
 // Protocols should be an array of strings that turn into Sec-WebSocket-Protocol.
-- (id)initWithURLRequest:(NSURLRequest *)request protocols:(NSArray *)protocols;
-- (id)initWithURLRequest:(NSURLRequest *)request;
+- (id)initWithURLRequest:(NSMutableURLRequest *)request protocols:(NSArray *)protocols allowSelfSignedCertificates:(BOOL)allowSelfSignedCertificates;
```

Comment:
This URL Request should be immutable, otherwise if someone has created an immutable `NSURLRequest` they'll have to call `mutableCopy` on it. This is going to be a breaking API change for a lot of people, and this isn't something that really needs to be mutable. If you yourself are trying to make a change to it under the hood, make a `mutableCopy`, make your changes, and assign it back to the ivar with `copy`.